students/carlos_novoa/lesson03/strformat_lab.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+"""
+Lesson3, String Formatting Lab Exercise
+"""
+
+
+def task1():
+    for i, item in enumerate((2, 123.4567, 10000, 12345.67)):
+        if i == 0:
+            # padded number
+            f = '{:0>3}'.format(item)
+            r = 'file_{}'.format(f)
+            _task_values.append(r)
+        if i == 1:
+            # floating, 2 decimals
+            r = '{0:.2f}'.format(item)
+            _task_values.append(r)
+        if i == 2:
+            # scientific notation, 2 decimals
+            r = '{0:.2e}'.format(item)
+            _task_values.append(r)
+        if i == 3:
+            # scientific notation, 3 figures
+            r = '{0:3.2e}'.format(item)
+            _task_values.append(r)
+
+    # file_002 :   123.46, 1.00e+04, 1.23e+04
+    print(''.join(_task_values[:1]) + ' :   ' + ', '.join(_task_values[1:]))
```

Comment:
this can be a simple one line if you combine formatting done in the for loop:
`print("file_{:03d} : {:10.2f}, {:.2e}, {:.3g}".format(2, 123.4567, 10000, 12345.67))`